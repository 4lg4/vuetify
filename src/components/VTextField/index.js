import VTextField from './VTextField'
import VTextInput from './VTextInput'

/* istanbul ignore next */
VTextField.install = function install (Vue) {
  Vue.component(VTextField.name, VTextField)
  Vue.component(VTextInput.name, VTextInput)
}

export default VTextField
