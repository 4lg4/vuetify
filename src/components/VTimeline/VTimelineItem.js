import { inject as RegistrableInject } from '../../mixins/registrable'

import Colorable from '../../mixins/colorable'

import VIcon from '../VIcon'

export default {
  name: 'v-timeline-item',

  mixins: [Colorable, RegistrableInject('timeline', 'v-timeline', 'v-timeline-item')],

  props: {
    icon: {
      type: String,
      default: 'event'
    },
    noIcon: {
      type: Boolean,
      default: false
    },
    lineColor: { type: String },
    iconFillColor: { type: String }
  },

  computed: {
    classes () {
      if (this.lineColor) {
        return this.addTextColorClassChecks({}, this.lineColor)
      }

      return ''
    },
    iconFillColorClasses () {
      return this.iconFillColor ? this.iconFillColor : this.$parent.$props.iconFillColor ? this.$parent.$props.iconFillColor : null
    }
  },

  methods: {
    genBody () {
      return this.$createElement(
        'div',
        {
          ref: 'body',
          class: 'timeline__item--body'
        },
        this.$slots.item
      )
    },
    genIconHeader () {
      const icon = this.noIcon || this.$parent.$props.noIcon ? null : this.$slots.icon || this.$createElement(VIcon, this.icon)
      return this.$createElement(
        'div',
        {
          staticClass: 'timeline__item--head'
        },
        [
          this.$createElement(
            'div',
            {
              staticClass: 'timeline__item--head-icon',
              class: this.iconFillColorClasses
            },
            [icon]
          )
        ]
      )
    }
  },

  render (h) {
    const children = []

    children.push(this.genIconHeader())
    children.push(this.genBody())

    return h(
      'li',
      {
        staticClass: 'timeline__item',
        class: this.classes
      },
      children
    )
  }
}
