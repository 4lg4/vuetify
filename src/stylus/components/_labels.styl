@import '../bootstrap'
@import '../theme'

v-label($material)
  color: $material.text.secondary

theme(v-label, "v-label")

.v-label
  font-size: 16px
  line-height: 1
  transition: .3s $transition.swing
