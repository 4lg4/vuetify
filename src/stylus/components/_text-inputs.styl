@import '../bootstrap'
@import '../theme'

v-text-input($material)
  border-bottom: 1px solid $material.dividers

  input
    color: $material.text.primary

  &--is-disabled
    .v-input__label,
    input
      color: $material.text.disabled

    &:before
      background-image: linear-gradient(
        to right,
        $material.text.disabled 0,
        $material.text.disabled 33%,
        transparent 0
      )

theme(v-text-input, "v-text-input")

.v-text-input
  align-items: flex-end
  color: inherit
  display: flex
  margin-top: 16px
  position: relative
  width: 100%

  &:before,
  &:after
    bottom: -1px
    content: ''
    height: 2px
    left: 0
    position: absolute
    transition: $primary-transition
    width: 100%

  &:before
    background-color: transparent
    background-position: bottom
    background-repeat: repeat-x
    background-size: 3px 1px

  &:after
    background-color: currentColor
    transform: scaleX(0)

  input
    flex: 1 1 auto
    font-size: 16px
    height: 32px
    padding: 8px 0 8px
    line-height: 32px
    width: 100%

    &:focus,
    &:active
      outline: none

  .v-label
    position: absolute
    min-height: 8px
    pointer-events: none
    position: absolute
    top: calc(50% - 8px)
    transform-origin: top left
    transition: .3s $transition.swing

    &--active
      transform: translate(0, -19px) scale(.75)

  &--is-focused:after
      transform: scaleX(1)
