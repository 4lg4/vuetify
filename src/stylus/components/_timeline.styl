@import "../bootstrap"
@import "../theme"

/* Themes */
timeline($material)
  background-color: $material.timelines
  color: $material.text.primary

theme(timeline, "timeline")

.timeline
  max-width: 1104px
  margin: 0 auto
  padding: 0 30px 0 30px
  overflow: hidden

  &--hover
    cursor: pointer
    transition: all 0.4s cubic-bezier(0.25, 0.8, 0.25, 1)
    transition-property: box-shadow

    &:hover
      elevation(8)

  &--raised
    elevation(3)

  &__container
    margin: 30px 0 0 0
    padding: 0
    list-style: none
    position: relative

    &> .timeline__item
      position: relative
      min-height: 128px

      & .timeline__line
        content: ""
        position: absolute
        top: 44px
        padding: 0
        bottom: 0
        background: currentColor
        left: 10%

      &:last-child .timeline__line
        position: relative

        &:before
          width: 0px

.timeline__container > .timeline__item
  & .timeline__item--body__container
    margin: 0 0 0 15%
    padding-bottom: 48px
    padding-top: 24px
    position: relative

  & .timeline__icon
    width: 44px
    height: 44px
    speak: none
    line-height: 44px
    position: absolute
    border-radius: 50%
    background: inherit
    box-shadow: currentColor 0px 0px 0px 8px 
    text-align: center
    left: 10%
    top: 0
    padding: 0
    margin: 0 0 0 -28px

@media screen and (max-width: $grid-breakpoints.sm)
  .timeline
    padding: 0 15px 0 15px

    &__container
      margin: 16px 0 0 0

      &> .timeline__item .timeline__line
        left: 6%

  .timeline__container > .timeline__item
    & .timeline__item--body__conatainer
      margin: 0 0 0 12%

    & .timeline__icon
      left: 6%
