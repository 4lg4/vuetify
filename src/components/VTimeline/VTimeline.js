import '../../stylus/components/_timeline.styl'

//  MIXINS
import Colorable from '../../mixins/colorable'

export default {
  name: 'v-timeline',

  mixins: [Colorable],

  provide () {
    return {
      right: this.right,
      left: this.left,
      alternate: this.alternate
    }
  },

  props: {
    right: {
      type: Boolean,
      default: false
    },
    left: {
      type: Boolean,
      default: false
    },
    alternate: {
      type: Boolean,
      default: true
    }
  },

  render (h) {
    return h(
      'ul',
      {
        staticClass: 'v-timeline'
      },
      this.$slots.default
    )
  }
}
