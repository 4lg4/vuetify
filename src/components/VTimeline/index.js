import VTimeline from './VTimeline'

export { VTimeline }

/* istanbul ignore next */
VTimeline.install = function install (Vue) {
  Vue.component(VTimeline.name, VTimeline)
}

export default VTimeline
