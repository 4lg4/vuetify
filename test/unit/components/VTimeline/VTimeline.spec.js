import { test } from '@/test'
import Vue from 'vue'
import { VTimeline, VTimelineItem } from '@/components/VTimeline'

test('VTimeline.js', ({ mount, compileToFunctions }) => {
  it('should render a div with class timeline', () => {
    const wrapper = mount(VTimeline)
    expect(wrapper.is('div')).toBe(true)
    expect(wrapper.hasClass('timeline')).toBe(true)
  })

  it('should render one ul with timeline__container class and match snapshot', () => {
    const wrapper = mount(VTimeline)
    const listElements = wrapper.find('.timeline ul')

    expect(listElements).toHaveLength(1)

    const ul = wrapper.first('ul')

    expect(ul.is('ul')).toBe(true)
    expect(ul.hasClass('timeline__container')).toBe(true)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render a light component', () => {
    const wrapper = mount(VTimeline, {
      propsData: {
        light: true
      }
    })
    const div = wrapper.find('.timeline')[0]
    expect(div.hasClass('theme--light')).toBe(true)
  })

  it('should render a dark component', () => {
    const wrapper = mount(VTimeline, {
      propsData: {
        dark: true
      }
    })
    const div = wrapper.find('.timeline')[0]
    expect(div.hasClass('theme--dark')).toBe(true)
  })

  it('should inject slot to children', () => {
    const { render } = compileToFunctions(`
      <v-timeline>
        <v-timeline-item/>
      </v-timeline>
    `)
    const component = Vue.component('test', {
      components: {
        VTimeline,
        VTimelineItem
      },
      render
    })
    const wrapper = mount(component)

    expect(wrapper.html()).toMatchSnapshot()
  })
})
