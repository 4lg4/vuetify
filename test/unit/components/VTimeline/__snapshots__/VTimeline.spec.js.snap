// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VTimeline.js should inject slot to children 1`] = `

<div class="timeline">
  <ul class="timeline__container">
    <li class="timeline__item">
      <div class="timeline__line grey--text text--lighten-2"
           style="width: 8px; margin-left: -0.66em;"
      >
      </div>
      <div class="timeline__item--head white grey--text text--lighten-2">
        <div class="timeline__icon white grey--text text--lighten-2"
             style="box-shadow: currentColor 0px 0px 0px 8px;"
        >
          <i aria-hidden="true"
             class="icon material-icons"
             style="font-size: 24px;"
          >
            event
          </i>
        </div>
      </div>
      <div class="timeline__item--body__container">
        <div class="timeline__item--body">
        </div>
      </div>
    </li>
  </ul>
</div>

`;

exports[`VTimeline.js should render one ul with timeline__container class and match snapshot 1`] = `

<div class="timeline">
  <ul class="timeline__container">
  </ul>
</div>

`;
